### PR TITLE
Add a |textlayerrendered| event

### PR DESCRIPTION
**Pull request by Snuffleupagus**
Tentative patch, implementing https://github.com/mozilla/pdf.js/pull/5511#issuecomment-68437309.

### CODE REVIEW

File: web/pdf_viewer.js [hunk 1]
```diff
@@ -250,6 +250,14 @@ var PDFViewer = (function pdfViewer() {
           });
           self.container.dispatchEvent(event);
         };
+        pageView.textLayerOnAfterDraw =
+            function pdfViewLoadTextLayerOnAfterDraw() {
+          var event = document.createEvent('CustomEvent');
+          event.initCustomEvent('textlayerrendered', true, true, {
+            pageNumber: pageView.id
+          });
+          self.container.dispatchEvent(event);
```
Comment (yurydelendik) [inline web/pdf_viewer.js]: I think you can move event dispatching to the text_layer_builder.js (and avoid callback stuff)

Comment (Snuffleupagus) [inline web/pdf_viewer.js]: I'm not opposed to implementing this the way you describe (should be fewer lines of code).

The reason that I didn't do so, is that the `pagerendered` event is placed in pdf_viewer.js and not in pdf_page_view.js. Hence I figured that for the sake of consistency, `textlayerrendered` should be implemented similarly.

Comment (yurydelendik) [inline web/pdf_viewer.js]: I see, shall we move pagerendered event too? :)

File: web/text_layer_builder.js [hunk 2]
```diff
@@ -53,6 +54,17 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
   }
 
   TextLayerBuilder.prototype = {
+    _afterRender: function TextLayerBuilder_afterRender() {
+      if (!this.renderingDone) {
```
Comment (yurydelendik) [inline web/text_layer_builder.js]: you can remove this. or make it 'finishRendering' and assign `this.renderingDone = true;` here

Comment (Snuffleupagus) [inline web/text_layer_builder.js]: Fixed; I went for the `_finishRendering` option since that meant less code duplication.

**Comment by yurydelendik** [comment]
Thank you

**Pull request merged**